torch_geometric/data/hetero_data.py

```diff
@@ -487,6 +487,149 @@ def validate(self, raise_on_error: bool = True) -> bool:
 
         return status
 
+    def separate(self) -> List[Self]:
```

Comment:
```suggestion
    def connected_components(self) -> List[Self]:
```
Nit: rename to connected_components.